lib/ansible/galaxy/role.py

```diff
@@ -207,10 +207,73 @@ def fetch(self, role_data):
                 temp_file.close()
                 return temp_file.name
             except Exception as e:
-                display.error(u"failed to download the file: %s" % to_text(e))
+                display.error("failed to download the file: %s" % to_text(e))
 
         return False
 
+    @staticmethod
+    def _is_path_sane(path: str) -> bool:
+        """
+        Check if the given path from a tar member is safe to extract.
+
+        This method looks for common risks that can lure in the path:
+        - Absolute paths that allow to create files outside of the target.
+        - `..` parts that allow to write files outside of the target.
+
+        It does NOT check:
+        - Symbols that are not allowed in environments (like `|` in Windows).
+        - Embedded variables like `$HOME/.config`. Please do not pass paths to a shell.
+        """
+
+        if path.startswith('/'):
```

Comment:
Absolute paths still can contain `..` in the middle of the path and that would be processed when accessing paths.

Reply:
https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extractall